### PR TITLE
Add pip install tests for different Python versions

### PR DESCRIPTION
**Pull request by kennysong**
This PR adds GitHub Actions to test installing LangCheck on this combination of environments:

```
        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
        os: [ubuntu-latest, windows-latest, macos-14]
```

It also adds a "Installation FAQ" in the documentation with instructions on how to fix some issues we've seen.

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -18,19 +18,32 @@ jobs:
...
-          python-version: '3.8.16'
+          python-version: '3.8'
 
       # Install the langcheck package with dev dependencies
       - name: Install
         run: |
           pip install --upgrade pip
           pip install .[dev]
 
+      # Remove unneeded system libraries to maximize disk space
```
Comment (kennysong) [inline .github/workflows/pytest.yml:34]: This is also in https://github.com/citadel-ai/langcheck/pull/84

File: README.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ Simple, Pythonic building blocks to evaluate LLM applications.
 pip install langcheck
 ```
 
+Having installation issues? [See the FAQ](https://langcheck.readthedocs.io/en/latest/installation.html#installation-faq).
```
Comment (kennysong) [inline README.md:28]: @liwii can you add this link in README_ja.md as well?

**Comment by kennysong** [comment]
@liwii Btw this is ready for review!

**Comment by liwii** [comment]
Let me take a look!!

File: README.md (hunk 2 above)
Comment (liwii) [reply README.md:28]: Added the link to the English FAQ.

**Review by liwii — APPROVED** [review]
Overall LGTM, let's work on splitting the per-language dependencies soon.

**Comment by kennysong** [comment]
Thanks and sounds good!

**Pull request merged**